drivers/crypto/crypto_rts5912_sha.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Realtek Semiconductor Corporation, SIBG-SD7
+ *
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_sha
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/init.h>
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sha256_hw_rtk, CONFIG_CRYPTO_LOG_LEVEL);
+
+#include "reg/reg_crypto.h"
+#include "crypto_rts5912_priv.h"
+
+static struct rts5912_sha256_context rts5912_sha256_ctx;
+
+#define RTS5912_SHA2_REG_BASE             ((SHA2_Type *)DT_INST_REG_ADDR(0))
+#define RTS5912_SHA2DMA_REG_BASE          ((SHA2DMA_Type *)DT_INST_PROP(0, sha2dma_reg))
```

Comment:
Follow the convention of other Zephyr drivers, create a `const` configuration structure to store these pointers and pass the pointer to the config to the `DEVICE_DT_INST_DEFINE` call

```
struct rts5912_sha_config {
    volatile struct sha_type *sha2_regs;
    volatile struct sha2dma_type *sha2dma_regs;
};

const struct rts5912_sha_config cfg_0 = {
    .sha2_regs = DT_INST_REG_ADDR(0),
    .sha2dma_regs = DT_INST_PROP(0, sha2dma_reg),
};

DEVICE_DT_INST_DEFINE(0, &rts5912_sha_init, NULL, NULL, &cfg_0, POST_KERNEL,

```